### ISSUE TITLE
ConcurrentModificationException will be reported when concurrent

### ISSUE THREAD

**Issue by Jude95** [ReactiveCircus/cache4k#23]
```
java.util.ConcurrentModificationException
        java.base/java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:756)
        java.base/java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:778)
        io.github.reactivecircus.cache4k.RealCache.expireEntries(RealCache.kt:174)
        io.github.reactivecircus.cache4k.RealCache.put(RealCache.kt:120)
```

**Comment by Shabinder** [comment]
+1:

```
java.util.ConcurrentModificationException at 
java.util.LinkedHashMap$LinkedHashIterator.nextNode(LinkedHashMap.java:760) at
java.util.LinkedHashMap$LinkedKeyIterator.next(LinkedHashMap.java:782) at
kotlin.collections.CollectionsKt___CollectionsKt.firstOrNull(_Collections.kt:272) at
io.github.reactivecircus.cache4k.RealCache.evictEntries(RealCache.kt:206) at
io.github.reactivecircus.cache4k.RealCache.put(RealCache.kt:140)
```

**Comment by Shabinder** [comment]
let cacheBuilder optionally accept a parameter to be used as a map underhood

**Comment by ychescale9** [comment]
I think we need to revert back to using `stately` which is thread-safe on JVM.

**Comment by Shabinder** [comment]
@ychescale9 will you be making a release anytime soon ?

**Comment by ychescale9** [comment]
I'll try to find some time in the coming weeks.

**Comment by luca992** [comment]
So I'm having [issues](https://github.com/Kamel-Media/Kamel/issues/75#issuecomment-1959271480) with cache4k and `ConcurrentModificationException`s caused by [stately](https://github.com/touchlab/Stately/issues/105) on jvm.

Ktor's `io.ktor.util.collections.ConcurrentMap` seems to be stable, maybe their implementation could be used instead?

**Comment by ychescale9** [comment]
Thanks, will look into it.

**Comment by hoc081098** [comment]
+1 for Ktor Utils Collections 😊

**Comment by ychescale9** [comment]
Just released 0.13.0 with `ConcurrentHashMap` on JVM. Please let me know if this fixes the issue.
